SabreTools.Serialization/Wrappers/ISO9660.Extraction.cs

```diff
@@ -81,32 +81,31 @@ private bool ExtractExtent(int extentLocation, Dictionary<int, int> extractedFil
                         // Append directory name
                         string outDirTemp = Path.Combine(outputDirectory, encoding.GetString(dr.FileIdentifier));
                         if (includeDebug) Console.WriteLine($"Extracting to directory: {outDirTemp}");
+
+                        // Ensure directory exists
+                        var directoryName = Path.GetDirectoryName(outDirTemp);
+                        if (directoryName != null && !Directory.Exists(directoryName))
+                            Directory.CreateDirectory(directoryName);

```

Comment:
Since you're always checking the directory within the `ExtractFile` method, you can avoid this additional code.